src/main/resources/com/puppycrawl/tools/checkstyle/grammar/java.g

```diff
@@ -386,8 +394,8 @@ builtInTypeSpec[boolean addImagNode]
 // A type name. which is either a (possibly qualified and parameterized)
 // class name or a primitive (builtin) type
 type
-    :    classOrInterfaceType[false]
-    |    builtInType
+    :    ({LA(1) == AT}? annotations | )
```

Comment:
note: `type` is only used in `newExpression`

reference: in the spec, [`PrimitiveType`](https://docs.oracle.com/javase/specs/jls/se8/html/jls-4.html#jls-PrimitiveType) allows an annotation (similar to [`ClassOrInterfaceTypeToInstantiate`](https://docs.oracle.com/javase/specs/jls/se8/html/jls-15.html#jls-ClassOrInterfaceTypeToInstantiate))
```
PrimitiveType:
  {Annotation} NumericType
  {Annotation} boolean
```
but while checkstyle's `classOrInterfaceType` allows an annotation, `builtInType` by itself doesn't. 

My thinking was to allow the annotation here in `type` for either case, rather than update `builtInType` which is used in several other places, so hoping to minimize changes.